src/CMakeLists.txt

```diff
@@ -48,12 +48,13 @@ ENDIF()
 # Main library.
 ADD_LIBRARY(roboptim-core SHARED ${HEADERS} ${ROBOPTIM_CORE_SRC})
 TARGET_COMPILE_OPTIONS(roboptim-core PUBLIC "-Droboptim_core_EXPORTS")
-PKG_CONFIG_USE_DEPENDENCY(roboptim-core eigen3)
 PKG_CONFIG_USE_DEPENDENCY(roboptim-core liblog4cxx)
 
 # Add required libs to pkg-config file.
 SET(ROBOPTIM_API_BOOST_LIBRARIES date_time system filesystem)
-PKG_CONFIG_APPEND_BOOST_LIBS(${ROBOPTIM_API_BOOST_LIBRARIES})
+IF(NOT WIN32)
+  PKG_CONFIG_APPEND_BOOST_LIBS(${ROBOPTIM_API_BOOST_LIBRARIES})
+ENDIF(NOT WIN32)
```

Comment:
The macro relies on `${Boost_${COMPONENT}_LIBRARY}` so providing the full library name does not need to be done manually. It would append `boostxxx-vcCompilerVersion-multithread-BoostVersion.lib` if this is what `FIND_PACKAGE(Boost ...)` returned.

As to whether or not these libraries in `roboptim-core.pc` are relevant, it was either 1) letting the end-user deal with these extra dependencies without necessarily knowing what's needed until the code is compiled and the linker returns an error (e.g. using `OptimizationLogger` without linking with `boost_system etc.`), or 2) taking a preemptive step with the risk that the Boost libraries won't be needed after all. I took the second option as it's easier for users and more resilient to internal changes to `roboptim-core` (e.g. if we start using something other than `boost::filesystem` for the paths). `libroboptim-core.so` is not linked with Boost `filesystem/system/date_time` as it does not need to be (unless we use explicit template instantiations), and these hypothetical dependencies are propagated through the pkg-config file instead.

For example, this works directly:
```cpp
/***********
*  foo.cc  *
***********/
#include <roboptim/core/optimization-logger.hh>

int main(int argc, char *argv[])
{
  using namespace roboptim;
  OptimizationLogger<Solver<EigenMatrixDense> >* logger;
  return 0;
}
```
```sh
$ clang++ foo.cc $(pkg-config --libs roboptim-core) $(pkg-config --cflags roboptim-core)
```

Now, we simply need to make sure that the same assumptions can be made for the Windows builds.